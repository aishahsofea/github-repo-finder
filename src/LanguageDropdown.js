import React from "react";
import InputLabel from "@material-ui/core/InputLabel";
import MenuItem from "@material-ui/core/MenuItem";
import FormControl from "@material-ui/core/FormControl";
import Select from "@material-ui/core/Select";

export class LanguageDropdown extends React.Component {
  constructor() {
    super();
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(event) {
    this.props.handleLanguage(event.target.value);
  }

  render() {
    const languages = [
      "Python",
      "JavaScript",
      "Java",
      "C#",
      "C",
      "C++",
      "Go",
      "R",
      "Swift",
      "PHP",
    ];
    return (
      <FormControl
        style={{
          marginRight: "50px",
          marginLeft: "50px",
          minWidth: 200,
          textAlign: "left",
        }}
      >
        <InputLabel>Language</InputLabel>
        <Select
          onChange={(event) => this.handleChange(event)}
          value={this.props.language}
        >
          <MenuItem value={null}>{null}</MenuItem>
          {languages.map((lang) => (
            <MenuItem value={lang} key={lang}>
              {lang}
            </MenuItem>
          ))}
        </Select>
      </FormControl>
    );
  }
}
