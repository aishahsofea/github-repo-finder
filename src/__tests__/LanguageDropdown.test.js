import React from "react";
import { LanguageDropdown } from "../LanguageDropdown";
import { shallow, configure } from "enzyme";
import * as Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

describe("<LanguageDropdown />", () => {
  it("must render correctly", () => {
    const wrapper = shallow(<LanguageDropdown />);
    expect(wrapper).toMatchSnapshot();
  });
});
