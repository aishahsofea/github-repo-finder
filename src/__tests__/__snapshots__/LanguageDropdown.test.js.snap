// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LanguageDropdown /> must render correctly 1`] = `ShallowWrapper {}`;
